tests/network/probes.go

```diff
@@ -146,15 +146,11 @@ var _ = SIGDescribe("[ref_id:1182]Probes", func() {
 			table.Entry("[test_id:6741]status change with guest-agent availability", guestAgentPingProbe, blankIPFamily, true, true),
 		)
 
-		table.DescribeTable("should fail", func(readinessProbe *v1.Probe, isExecProbe bool) {
+		table.DescribeTable("should fail", func(readinessProbe *v1.Probe, vmiFactory func(opts ...libvmi.Option) *v1.VirtualMachineInstance) {
 			By(specifyingVMReadinessProbe)
-			if isExecProbe {
-				vmi = tests.NewRandomFedoraVMIWithGuestAgent()
-				vmi.Spec.ReadinessProbe = readinessProbe
-				vmi = tests.VMILauncherIgnoreWarnings(virtClient)(vmi)
-			} else {
-				vmi = createReadyCirrosVMIWithReadinessProbe(virtClient, readinessProbe)
-			}
+			vmi = vmiFactory()
+			vmi.Spec.ReadinessProbe = readinessProbe
+			vmi = tests.VMILauncherIgnoreWarnings(virtClient)(vmi)
```

Comment:
If I am not mistaken, it is suppose not to successfully start.
On the other hand, I pretty much lost it when seeing conditions and phases of a VMI in the same "room".

Reply:
> Why calling `tests.WaitForSuccessfulVMIStartIgnoreWarnings` is not enough?

I don't know why both `tests.WaitForSuccessfulVMIStartIgnoreWarnings` and `tests.VMILauncherIgnoreWarnings` exist, nor which of the two is prettier (I like the fact that I don't need to mention timeout, but also that I don't need to pass the virtClient, I care about neither). I decided to keep `tests.VMILauncherIgnoreWarnings` only because this is what was used in this test earlier. I intended to have a more thorough discussion about pros and cons of each launch method in another PR.

Reply:
I'm ok with another PR discussing it.